drivers/can/socket_can_generic.h

```diff
@@ -78,20 +78,21 @@ static inline int socket_can_setsockopt(struct device *dev, void *obj,
 					const void *optval, socklen_t optlen)
 {
 	struct socket_can_context *socket_context = dev->driver_data;
+	struct zcan_filter filter;
 	int ret;
 
 	if (level != SOL_CAN_RAW && optname != CAN_RAW_FILTER) {
 		errno = EINVAL;
 		return -1;
 	}
 
-	if (optlen != sizeof(struct can_filter)) {
```

Comment:
I would suggest an ASSERT optlen == sizeof(struct can_filter) here.

Reply:
We can add assert, no problem. It is just that the value is already checked also in sockets_can.c so we should not get into this function with wrong size address.

I noticed that we need to change the sending of the packet too, so "struct can_frame" should be used in sample when sending, then we need to convert it to "struct zcan_frame" as the driver is expecting that. I will add a commit for this in this PR.